In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
from sklearn.model_selection import train_test_split
import tensorflow as tf
from tensorflow import keras

df = pd.read_csv("C:/Users/nkm11/Desktop/AI_class/breast_cancer.csv", index_col=0)

In [2]:
X=df.drop('label',axis=1)
X.head()

mean radius  mean texture  mean perimeter  mean area  mean smoothness  \
0        17.99         10.38          122.80     1001.0          0.11840   
1        20.57         17.77          132.90     1326.0          0.08474   
2        19.69         21.25          130.00     1203.0          0.10960   
3        11.42         20.38           77.58      386.1          0.14250   
4        20.29         14.34          135.10     1297.0          0.10030   

   mean compactness  mean concavity  mean concave points  mean symmetry  \
0           0.27760          0.3001              0.14710         0.2419   
1           0.07864          0.0869              0.07017         0.1812   
2           0.15990          0.1974              0.12790         0.2069   
3           0.28390          0.2414              0.10520         0.2597   
4           0.13280          0.1980              0.10430         0.1809   

   mean fractal dimension  ...  worst radius  worst texture  worst perimeter  \
0                 0.07871  ...         25.38          17.33           184.60   
1                 0.05667  ...         24.99          23.41           158.80   
2                 0.05999  ...         23.57          25.53           152.50   
3                 0.09744  ...         14.91          26.50            98.87   
4                 0.05883  ...         22.54          16.67           152.20   

   worst area  worst smoothness  worst compactness  worst concavity  \
0      2019.0            0.1622             0.6656           0.7119   
1      1956.0            0.1238             0.1866           0.2416   
2      1709.0            0.1444             0.4245           0.4504   
3       567.7            0.2098             0.8663           0.6869   
4      1575.0            0.1374             0.2050           0.4000   

   worst concave points  worst symmetry  worst fractal dimension  
0                0.2654          0.4601                  0.11890  
1                0.1860          0.2750                  0.08902  
2                0.2430          0.3613                  0.08758  
3                0.2575          0.6638                  0.17300  
4                0.1625          0.2364                  0.07678  

[5 rows x 30 columns]

In [3]:
y=df['label']
y.value_counts()

label
1    357
0    212
Name: count, dtype: int64

In [4]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X, y,test_size=0.25,random_state=0)

## 모델

## LR

In [5]:
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC
from sklearn.neural_network import MLPClassifier
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import accuracy_score
from sklearn.metrics import confusion_matrix

In [6]:
# Build a logistic regression classifier and predict

# 모델 생성
clf_lr = LogisticRegression(random_state=0)
# 모델 훈련
clf_lr.fit(X_train, y_train)

# 예측된 값
pred_lr = clf_lr.predict(X_test)

print ("\n--- Logistic Regression Classifier ---")
print (accuracy_score(y_test, pred_lr))
print (confusion_matrix(y_test, pred_lr))


--- Logistic Regression Classifier ---
0.951048951048951
[[52  1]
 [ 6 84]]


C:\Users\nkm11\anaconda3\envs\dr\Lib\site-packages\sklearn\linear_model\_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


51과 84가 정답. 1과 6은 잘못 분류된 결과값이기에, 정확도가 0.951048951048951로 출력

### 우리가 해야 할 일 정리
1. 데이터 load 및 전처리
2. train과 test로 나눔 / xtrain, xtest, ytrain, ytest 총 4개로 나뉨
3. 모델 생성
4. 모델 훈련
5. 예측 진행 -> 출력하여 확인

데이터 분석이 어려운거지 이건 쉽고, SVM은 권하지 않는다(시간 오래걸린다).
기본적으로 SVM, RF 등 간단히 진행할것임.

In [8]:
# Build a decision tree classifier and predict

clf_dt = DecisionTreeClassifier(random_state=0)
clf_dt.fit(X_train, y_train)

pred_dt = clf_dt.predict(X_test)

print ("\n--- Decision Tree Classifier ---")
print (accuracy_score(y_test, pred_dt))
print (confusion_matrix(y_test, pred_dt))


--- Decision Tree Classifier ---
0.8811188811188811
[[50  3]
 [14 76]]


In [9]:
clf_nn = MLPClassifier(random_state=0)
clf_nn.fit(X_train, y_train)

pred_nn = clf_nn.predict(X_test)

print ("\n--- Neural Network Classifier ---")
print (accuracy_score(y_test, pred_nn))
print (confusion_matrix(y_test, pred_nn))


--- Neural Network Classifier ---
0.9300699300699301
[[48  5]
 [ 5 85]]


In [10]:
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score
import pandas as pd
import warnings
warnings.filterwarnings('ignore')
print ("\n--- Radom Forest ---")
rf_clf = RandomForestClassifier(random_state=0)
rf_clf.fit(X_train, y_train)
pred = rf_clf.predict(X_test)
print(accuracy_score(y_test,pred))
print (confusion_matrix(y_test, pred))


--- Radom Forest ---
0.972027972027972
[[52  1]
 [ 3 87]]


# 추가학습 (diabetes.csv)

In [14]:
df1 = pd.read_csv("C:/Users/nkm11/Desktop/AI_class/diabetes.csv", index_col=0)
df1

Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
Pregnancies                                                         
6                148             72             35        0  33.6   
1                 85             66             29        0  26.6   
8                183             64              0        0  23.3   
1                 89             66             23       94  28.1   
0                137             40             35      168  43.1   
...              ...            ...            ...      ...   ...   
10               101             76             48      180  32.9   
2                122             70             27        0  36.8   
5                121             72             23      112  26.2   
1                126             60              0        0  30.1   
1                 93             70             31        0  30.4   

             DiabetesPedigreeFunction  Age  Outcome  
Pregnancies                                          
6                               0.627   50        1  
1                               0.351   31        0  
8                               0.672   32        1  
1                               0.167   21        0  
0                               2.288   33        1  
...                               ...  ...      ...  
10                              0.171   63        0  
2                               0.340   27        0  
5                               0.245   30        0  
1                               0.349   47        1  
1                               0.315   23        0  

[768 rows x 8 columns]

In [13]:
X1=df1.drop('Outcome',axis=1)
X1.head()

Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
Pregnancies                                                         
6                148             72             35        0  33.6   
1                 85             66             29        0  26.6   
8                183             64              0        0  23.3   
1                 89             66             23       94  28.1   
0                137             40             35      168  43.1   

             DiabetesPedigreeFunction  Age  
Pregnancies                                 
6                               0.627   50  
1                               0.351   31  
8                               0.672   32  
1                               0.167   21  
0                               2.288   33

In [15]:
y1=df1['Outcome']
y1.value_counts()

Outcome
0    500
1    268
Name: count, dtype: int64

In [16]:
from sklearn.model_selection import train_test_split

X1_train, X1_test, y1_train, y1_test = train_test_split(X1, y1,test_size=0.25,random_state=0)

In [17]:
# Build a logistic regression classifier and predict

# 모델 생성
clf_lr = LogisticRegression(random_state=0)
# 모델 훈련
clf_lr.fit(X1_train, y1_train)

# 예측된 값
pred_lr = clf_lr.predict(X1_test)

print ("\n--- Logistic Regression Classifier ---")
print (accuracy_score(y1_test, pred_lr))
print (confusion_matrix(y1_test, pred_lr))


--- Logistic Regression Classifier ---
0.7916666666666666
[[116  14]
 [ 26  36]]


# 과제 공지
1. 아나콘다 설치
2. 파이썬 설치
3. tensorflow 설치

세 가지 다시 해올 것.


1. 당뇨병 데이터셋으로 금일 진행한 내용 반복학습.

In [18]:
!pip list

Package                      Version
---------------------------- ---------
absl-py                      1.4.0
aiofiles                     22.1.0
aiosqlite                    0.18.0
anyio                        3.5.0
argon2-cffi                  21.3.0
argon2-cffi-bindings         21.2.0
asttokens                    2.0.5
astunparse                   1.6.3
attrs                        22.1.0
Babel                        2.11.0
backcall                     0.2.0
beautifulsoup4               4.12.2
bleach                       4.1.0
Bottleneck                   1.3.5
brotlipy                     0.7.0
cachetools                   5.3.1
certifi                      2023.7.22
cffi                         1.15.1
charset-normalizer           2.0.4
colorama                     0.4.6
comm                         0.1.2
contourpy                    1.1.0
cryptography                 41.0.2
cycler                       0.11.0
debugpy                      1.6.7
decorator                    5.1.1


이후 딥러닝 Express 이론 요약진행.

github, 2, 3번 예시로, git 내용으로 라이브러리들이 뭘 하는지 간단히 설명해주심.

Error 메시지는 읽어보고 해결하자. 보통은 설치 문제.

# 제일 중요한 것 -> 93 페이지(시험)
1. classification(분류)
2. regrassion(회귀)
3. MSE(평균제곱오차) 예측값과 실제값의 차이를 제곱하여 평균낸 값. 모델의 예측이 실제값에서 얼마나 벗어났는지를 나타내며, 값이 작을수록 모델의 성능이 좋다

97,107, 109, k-means만, 110, 111, 